clientv3/watch.go

```diff
@@ -497,9 +497,20 @@ func (w *watcher) resume() (ws pb.Watch_WatchClient, err error) {
 	return ws, v3rpc.Error(err)
 }
 
+func (w *watcher) checkClosed() bool {
+	w.rc.client.mu.RLock()
+	closed := w.rc.client.cancel == nil
+	w.rc.client.mu.RUnlock()
+	return closed
+}
+
 // openWatchClient retries opening a watchclient until retryConnection fails
 func (w *watcher) openWatchClient() (ws pb.Watch_WatchClient, err error) {
 	for {
+		if w.checkClosed() {
```

Comment:
I think using w.rc.acquire() would fix the connection race with close/w.remote.Watch